package.json

```diff
@@ -294,8 +298,8 @@
             },
             {
                 "command": "language-julia.clearCurrentInlineResult",
-                "key": "Ctrl+I Ctrl+D",
-                "when": "editorTextFocus && editorLangId == julia"
+                "key": "Escape",
+                "when": "editorTextFocus && !editorHasSelection && !findWidgetVisible && !markerNavigationVisible && !parameterHintsVisible && !inSnippetMode && !isInEmbeddedEditor && !suggestWidgetVisible && !onTypeRenameInputVisible && !renameInputVisible && editorLangId == julia && juliaHasInlineResult"
```

Comment:
I have to admit that `Esc` does not strike me as a good keyboard shortcut for this... First because I'm super nervous that we might have overlooked some crucial `when` condition here that we need, and second because `Esc` just never does seem to remove anything in any editor. Should we try to find something a little safer?

Reply:
Well, `Escape` does clear the intellisense popup and a bunch more. Those are fairly similar in that they provide information at a certain spot in the editor -- inline results aren't part of the text document either.

We can also enumerate all conditions in VSCode core. The popular gitlens extension also [does something similar](https://github.com/eamodio/vscode-gitlens/blob/6cfd9fdedd7c6ec3bfa732af7c418bbbecdfba54/package.json#L5593-L5597).

Reply:
FWIW, I strongly feel that `Escape` is the right keybinding for this (as does @aviatesk, I suspect). If you really don't want this for whatever reason, we can however put this behind an option (easy to add to the `when` condition) and have a "safer" alternative keybinding enabled by default.

Reply:
To me there is a pretty strong difference between UI elements like intellisense popups and our annotations, I think our annotations feel very much like "part of the document", because they scroll with the doc, they stay there if you switch to a different tab and back etc. UI elements like context menus, where Esc is routinely used to dismiss them, feel very different, I think: they are very transitory and disappear with almost any UI interaction. In my mind the inline results feel much more like notebook output, which I think one would also consider part of the docs?

There seem to be a lot of extensions that add decorations of various form to documents, and as far as I can tell (maybe with the exception of GitLens), almost none of them use `Esc` as the "delete decorations" keyboard shortcut?

I think we should probably not start to enable disable keyboard shortcuts via config options: wouldn't it be easier for users to just create their own keyboard binding in that situation?

Reply:
So what's the concern here? That
1. it is too easy to accidentally dismiss a result, or
2. that the current definition might miss a condition, so that you might need to press Escape twice to do what you actually meant to do (first one for the result, second one for the original intent)?

The reason for potentially putting this behind a setting is that figuring out the correct `when` statement isn't trivial. Could of course put it into the docs.

FWIW, Juno has been using Escape for this since forever and I don't think we heard any complaints about it so far -- someone even asked for this keybinding in VSCode on slack the other day :)

Anyhow, do you have an alternative keybinding?

Reply:
I honestly don't have a strong opinion on this, but I'm pretty sure that there won't be an issue to go with <kbd>esc</kbd>, given there hasn't been any complaint on this in Juno.
Is there any good alternative ? <kbd>esc</kbd> is actually used for lots of things, like notifications/clearing popups/cancelling commands, etc, and to me, "removing inline results" can pretty naturally align with them

Reply:
So my main concern is that I think it is not really aligned with how `Esc` is used in the UI of VS Code and many other pieces of software. `Esc` in my mind makes temporary UI elements disappear/cancel. The decorations in my mind are part of the document, because they have a lifetime that for example extends beyond a tab switch in the VS Code window (and they survive other things like window minimize/maximize etc), so in my mind they are in a different bucket of UI element than say a context menu, an active edit in a tree view or something like that, where `Esc` is a typical cancel option.

I don't have a good alternative keybinding, but I find even the current keybinding a better choice.

Reply:
Alright, reverted to the old keybinding.

I still maintain that `Escape` is a good choice here, but that's not a hill I'm willing to die on :)

Reply:
We can revisit later as well, when we hopefully revisit the whole keybinding story!